lib_nbgl/src/nbgl_buttons.c

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file nbgl_buttons.c
+ * Implementation of buttons management in NBGL
+ */
+
+#ifndef HAVE_SE_TOUCH
+/*********************
+ *      INCLUDES
+ *********************/
+#include <string.h>
+#include "nbgl_debug.h"
+#include "nbgl_buttons.h"
+#include "nbgl_lns.h"
+#include "os_pic.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static uint8_t gButtonMask            = 0;
+static uint8_t gButtonSameMaskCounter = 0;
+
+/**********************
+ *      VARIABLES
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static nbgl_buttonEvent_t maskToEvent(uint8_t mask)
+{
+    nbgl_buttonEvent_t event = INVALID_BUTTON_EVENT;
+    switch (mask) {
+        case RELEASED_MASK | LEFT_BUTTON | RIGHT_BUTTON:
+            event = BUTTON_BOTH_PRESSED;
+            break;
+
+        case LEFT_BUTTON | RIGHT_BUTTON:
+            event = BUTTON_BOTH_TOUCHED;
+            break;
+
+        case CONTINUOUS_MASK | LEFT_BUTTON:
+            event = BUTTON_LEFT_CONTINUOUS_PRESSED;
+            break;
+
+        case RELEASED_MASK | LEFT_BUTTON:
+            event = BUTTON_LEFT_PRESSED;
+            break;
+
+        case CONTINUOUS_MASK | RIGHT_BUTTON:
+            event = BUTTON_RIGHT_CONTINUOUS_PRESSED;
+            break;
+
+        case RELEASED_MASK | RIGHT_BUTTON:
+            event = BUTTON_RIGHT_PRESSED;
+            break;
+    }
+
+    return event;
+}
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+/**
+ * @brief Function to be called periodically to check touchscreen state
+ * and coordinates
+ * @param buttonState state of both buttons (only 2 LSB are used)
+ */
+void nbgl_buttonsHandler(uint8_t buttonState)
+{
+    uint8_t button_mask;
+    uint8_t button_same_mask_counter;
+
+    // enable speeded up long push (continuous)
```

Comment:
Keeping the button pressed to review faster ?
Where does this come from ?

Reply:
This is there only because it's needed on LNX and LNSP and I kept the file really similar to the one in the ledger-secure-sdk branch.
Apps that only use the NBGL use case API won't use it. But some apps can still use it if they use the lower API.